File: SampleApps/ObjC/WTBank ObjC WatchKit Extension/InterfaceController.m
//
//  InterfaceController.m
//  WTBank ObjC WatchKit Extension
//
//  Created by Sam Ingle on 6/25/15.
//  Copyright (c) 2015 Webtrends. All rights reserved.
//

#import "InterfaceController.h"
#import "WebtrendsWatchKitSDK.h"

@interface InterfaceController()

@property (weak, nonatomic) IBOutlet WKInterfaceLabel *balanceLabel;
@property (nonatomic) NSInteger balance;
@end


@implementation InterfaceController

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];
    self.balance = 100;
}
- (IBAction)refreshBalance
{
    self.balance++;
    [self.balanceLabel setText:[NSString stringWithFormat:@"$%@", @(self.balance)]];
    
    WTEventMeta *meta = [WTEventMeta eventMetaForPath:nil description:@"InterfaceController" type:@"Tap" customParams:nil];
    [[WTWatchDataCollector sharedCollector] sendEventForAction:meta];
}

- (void)willActivate {
    // This method is called when watch view controller is about to be visible to user
    [super willActivate];
}

- (void)didDeactivate {
    // This method is called when watch view controller is no longer visible
    [super didDeactivate];
}

@end

File: SampleApps/Swift/WTBank Swift WatchKit Extension/InterfaceController.swift
//
//  InterfaceController.swift
//  WTBank Swift WatchKit Extension
//
//  Created by Sam Ingle on 6/26/15.
//  Copyright (c) 2015 Webtrends. All rights reserved.
//

import WatchKit
import Foundation

class InterfaceController: WKInterfaceController {
    @IBOutlet weak var balanceLabel: WKInterfaceLabel!
    
    var balance = 100

    @IBAction func refreshBalance() {
        balance++
        balanceLabel.setText("$\(balance)")
        let meta = WTEventMeta(eventPath: nil, description: "InterfaceController", type: "tap", customParams: nil)
        WTWatchDataCollector.sharedCollector().sendEventForAction(meta)
    }
    override func awakeWithContext(context: AnyObject?) {
        super.awakeWithContext(context)
        
        // Configure interface objects here.
    }

    override func willActivate() {
        // This method is called when watch view controller is about to be visible to user
        super.willActivate()
    }

    override func didDeactivate() {
        // This method is called when watch view controller is no longer visible
        super.didDeactivate()
    }

}
